addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -0,0 +1,387 @@
+                            _logger.info(
+                                "%s, maximum number of tries reached" % errorcodes.lookup(e.pgcode)
+                            )
+                            break
+                        tries += 1
+                        wait_time = 1
+                        _logger.info(
+                            "%s, retry %d/%d in %.04f sec..."
+                            % (
+                                errorcodes.lookup(e.pgcode),
+                                tries,
+                                MAX_TRIES_ON_CONCURRENCY_FAILURE,
+                                wait_time,
+                            )
+                        )
+                        time.sleep(wait_time)
+                if tries == -1:
+                    continue
+                if tries >= MAX_TRIES_ON_CONCURRENCY_FAILURE:
+                    break
+                
+                # Incrementally commit to release picks as soon as possible and
+                # allow serialisation error to propagate to respect priority
+                # order
+                self.env.cr.commit()
+                # Only count as reserved the number of pickings at mls
+                to_reserve -= len(mls.mapped("picking_id"))
+
+                if self:
+                    # Only process the specified pickings
+                    break
+            _logger.info("Reserving stock for picking type %r completed.", picking_type)
+        return
+    
+    def apply_drop_location_policy(self):
+        """Apply suggested locations to move lines
+        raise ValidationError: if the policy set does not have
+                               _allow_preprocess set
+        """
+        by_pack_or_single = lambda ml: ml.package_id.package_id or ml.package_id or ml.id
+
+        for pick in self:
+            self.check_policy_for_preprocessing(pick.picking_type_id.u_drop_location_policy)
+            # Group by pallet or package
+            for _pack, mls in pick.move_line_ids.groupby(by_pack_or_single):
+                locs = pick.get_suggested_locations(mls)
+                if locs:
+                    mls.write({"location_dest_id": locs[0].id})
+
+    def check_policy_for_preprocessing(self, policy):
+        """ "Check policy allows pre processing as not all polices
+        can be used in this way
+        """
+        func = getattr(self, "_get_suggested_location_" + policy, None)
+
+        if not hasattr(func, "_allow_preprocess"):
+            raise ValidationError(
+                _("This policy(%s) is not meant to be used in " "preprocessing") % policy
+            )
+    
+    def get_suggested_locations(self, move_line_ids, limit=None, sort=True):
+        """Dispatch the configured suggestion location policy to
+        retrieve the suggested locations
+        """
+        result = self.env["stock.location"]
+
+        # WS-MPS: use self.ensure_one() and don't loop (self is used in all but
```

Comment:
Would suggest to remove specific names, and leave only the comment.
I understand that has been like that in the module that you are porting, seen that in other parts too.